src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -860,8 +861,14 @@ public void updateFromTopic(Topic topic) {
                     lastModifiedTime = Coerce.toLong(topic);
                     break;
                 case PERSISTED_CURRENT_PROCESSING_FILE_HASH:
-                    //TODO: the scenario of upgrading from older version needs to be handled in next PR.
+                    // If upgrade from older version, then the fileHash does not exist but fileName exists, then
+                    // transfer from fileName to fileHash.
+                    String savedName = Coerce.toString(PERSISTED_CURRENT_PROCESSING_FILE_NAME);
```

Comment:
This isn't doing what you want. This will give you the string "currentProcessingFileName", which will always be non-null.